File: app/src/main/java/com/example/newbiechen/ireader/presenter/contract/ReadContract.java
package com.example.newbiechen.ireader.presenter.contract;

import com.example.newbiechen.ireader.model.bean.BookChapterBean;
import com.example.newbiechen.ireader.ui.base.BaseContract;
import com.example.newbiechen.ireader.widget.page.TxtChapter;

import java.util.List;

/**
 * Created by newbiechen on 17-5-16.
 */

public interface ReadContract extends BaseContract{
    interface View extends BaseContract.BaseView {
        void showCategory(List<BookChapterBean> bookChapterList);
        void finishChapter();
        void errorChapter();
    }

    interface Presenter extends BaseContract.BasePresenter<View>{
        void loadMixCategory(String bookMixId);
        void loadSourceCategory(String bookBookId, String bookMixId);
        void loadChapter(String bookId,List<TxtChapter> bookChapterList);
    }
}

File: app/src/main/java/com/example/newbiechen/ireader/presenter/contract/ReadContract.kt
package com.example.newbiechen.ireader.presenter.contract

import com.example.newbiechen.ireader.db.entity.BookChapter
import com.example.newbiechen.ireader.ui.base.BaseContract
import com.example.newbiechen.ireader.widget.page.TxtChapter

interface ReadContract{
    interface View : BaseContract.BaseView {
        fun showCategory(bookChapters: List<BookChapter>)
        fun finishChapter()
        fun errorChapter()
    }

    interface Presenter : BaseContract.BasePresenter<View> {
        fun loadMixCategory(bookMixId: String)
        fun loadSourceCategory(bookBookId: String, bookMixId: String)
        fun loadChapter(bookId: String, bookChapterList: List<TxtChapter>)
    }
}